app/models/manageiq/providers/terraform_enterprise/automation_manager/orchestration_stack.rb

```diff
@@ -0,0 +1,36 @@
+class ManageIQ::Providers::TerraformEnterprise::AutomationManager::OrchestrationStack < ManageIQ::Providers::ExternalAutomationManager::OrchestrationStack
+  belongs_to :configuration_script,         :foreign_key => :configuration_script_id
+  belongs_to :configuration_script_payload, :foreign_key => :configuration_script_base_id
+
+  def self.display_name(number = 1)
+    n_('Run (Terraform Enterprise)', 'Runs (Terraform Enterprise)', number)
+  end
+
+  def self.create_stack(workspace, options = {})
+    run_id = raw_create_stack(workspace, options)
+
+    create!(
+      :name                         => workspace.name,
+      :ext_management_system        => workspace.manager,
+      :ems_ref                      => run_id,
+      :status                       => "pending",
+      :configuration_script         => workspace,
+      :configuration_script_payload => workspace.parent
+    )
+  end
+
+  def self.raw_create_stack(workspace, options = {})
+    workspace.run(options)
+  rescue => err
+    _log.error("Failed to create run from workspace [#{workspace.name}]: #{err}")
+    raise MiqException::MiqOrchestrationProvisionError, err.to_s, err.backtrace
+  end
+
+  def self.db_name
+    'ConfigurationJob'
+  end
+
+  def raw_status
+    Status.new(status)
+  end
```

Comment:
This usually comes from a live API call but since we refresh these from the provider I'd rather have this straight from the database and let the periodic stack.refresh_ems that Automate does keep this up to date

```
>> OrchestrationStack.last.raw_status
=> #<ManageIQ::Providers::TerraformEnterprise::AutomationManager::OrchestrationStack::Status:0x00007efdc3dcb960
 @reason="The run has errored. This is a final state.",
 @status="errored">
```